projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts

```diff
@@ -488,138 +476,85 @@ describe('Integration Tests: WizardComponent with Mock Parent', () => {
-  it('should have a stepper that correctly moves forward (via the parent)', () => {
+    pressBackButton();
     expectFirstStep();
-    wizard.move(true);
-    expectNthStep(1);
   });
 
-  it('should have a stepper that correctly moves backward', () => {
-    expectFirstStep();
-    stepper.move(true);
-    expectNthStep(1);
-
-    stepper.move(false);
+  it('should correctly moving forward', () => {
+    const indexChangedSpy = spyOn(wizard.indexChange, 'emit');
     expectFirstStep();
-  });
+    setNextStepAvailable();
 
-  it('should have a stepper that correctly moves backward (via the parent)', () => {
-    expectFirstStep();
-    wizard.move(false);
-    expectFirstStep();
+    pressNextButton();
+    expectNthStep(1);
+    expect(indexChangedSpy).toHaveBeenCalledWith(wizard.index);
   });
 
-  it('should have a stepper that does not move out of bound on forward movement', () => {
+  it('should not move out of bound on forward movement', () => {
     moveToLastStep();
-    stepper.move(true);
-    expect(stepper.index).toEqual(stepper.items.length);
+    expect(wizard.index).toEqual(wizard.items.length - 1);
   });
 
-  it('should have a stepper that does not move out of bound on backward movement', () => {
-    expectFirstStep();
-    stepper.move(false);
-    expectFirstStep();
-  });
+  it('should save', () => {
+    const onSaveSpy = spyOn(wizard, 'save');
 
-  it('should have a not save functionality', () => {
-    const state = false;
-    spyOn(stepper, 'move').withArgs(state);
-    spyOnWizardEmitters(false, false, true);
-    wizard.save(state);
-    expect(wizard.savingChange.emit).toHaveBeenCalledWith(state);
-  });
+    moveToLastStep();
+    setNextStepAvailable();
+    pressSaveButton();
 
-  it('should have a save functionality', () => {
-    const state = true;
-    spyOnWizardEmitters(false, false, true);
-    wizard.save(state);
-    expect(wizard.isSaved).not.toEqual(state);
-    expect(wizard.savingChange.emit).toHaveBeenCalledWith(state);
+    expect(onSaveSpy).toHaveBeenCalled();
   });
 
-  it('should be able to close', () => {
-    spyOn(stepper, 'reset');
-    spyOnWizardEmitters(true, true, true);
-
-    wizard.closeDialog();
-    expectWizardClosed();
+  it('should not be able to save on any step before the last step', () => {
+    const saveButton = getNativeElementAsHTMLElement(saveButtonDeclaration);
+    expect(saveButton).toBeNull();
   });
 
-  it('should be closable 300ms after saving', fakeAsync(() => {
-    expectIsClosable(true);
-
-    moveToLastStep();
-    fixture.detectChanges();
-    expectLastStep();
-
+  it('should be closable after saving', () => {
     expectIsSaved(false);
     expectIsClosable(true);
 
-    const isCloseButtonDisabled = isElementDisabled(closeButtonDeclaration);
-    expect(isCloseButtonDisabled).toBeFalse();
-
+    moveToLastStep();
     pressSaveButton();
     expectIsClosable(true);
+  });
 
-    tick(300);
-    wizard.closable = true;
+  it('should be closed', () => {
```

Comment:
naming: should be closed on what trigger?

Reply:
Done